assets/aws/files/install-hardened.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+if [[ "${DEBUG:-false}" == "true" ]]; then
+    set -x
+fi
+
+# Set some curl options so that temporary failures get retried
+# More info: https://ec.haxx.se/usingcurl-timeouts.html
+CURL_OPTS="-L --retry 100 --retry-delay 0 --connect-timeout 10 --max-time 300"
+
+# Update packages
+dnf -y update
+
+# Install 
+#  - uuid used for random token generation,
+#  - python for certbot
+dnf install -y uuid python3
+
+# Install certbot 
+python3 -m venv /opt/certbot
+/opt/certbot/bin/pip install --upgrade pip
+/opt/certbot/bin/pip install certbot certbot-dns-route53
+ln -s /opt/certbot/bin/certbot /usr/bin/certbot
+
+# Create teleport user. It is helpful to share the same UID
+# to have the same permissions on shared NFS volumes across auth servers and for consistency.
+useradd -r teleport -u ${TELEPORT_UID} -d /var/lib/teleport
+# Add teleport to adm group to read and write logs
+usermod -a -G adm teleport
+
+# Setup teleport run dir for pid files
+mkdir -p /run/teleport/ /var/lib/teleport /etc/teleport.d
+chmod 0700 /var/lib/teleport
+chown -R teleport:adm /run/teleport /var/lib/teleport /etc/teleport.d/
+
+# Download and install teleport binaries
+pushd /tmp || exit
+# Install the FIPS version of Teleport if /tmp/teleport-fips is present
+if [ -f /tmp/teleport-fips ]; then
+    TARBALL_FILENAME="/tmp/files/teleport-ent-v${TELEPORT_VERSION}-linux-amd64-fips-bin.tar.gz"
+    # Use a Teleport artifact uploaded from the build machine, if present
+    if [ -f ${TARBALL_FILENAME} ]; then
+        echo "Found locally uploaded Enterprise FIPS tarball ${TARBALL_FILENAME}, moving to /tmp/teleport.tar.gz"
+        mv ${TARBALL_FILENAME} /tmp/teleport.tar.gz
+    else
+        echo "Installing Enterprise Teleport version ${TELEPORT_VERSION} with FIPS support"
+        curl ${CURL_OPTS} -o teleport.tar.gz https://get.gravitational.com/teleport/${TELEPORT_VERSION}/teleport-ent-v${TELEPORT_VERSION}-linux-amd64-fips-bin.tar.gz
+    fi
+    tar -xzf teleport.tar.gz
+    cp teleport-ent/tctl teleport-ent/tsh teleport-ent/teleport /usr/local/bin
+    rm -rf /tmp/teleport.tar.gz /tmp/teleport-ent
+    # add --fips to 'teleport start' commands in FIPS mode
+    sed -i -E "s_ExecStart=/usr/local/bin/teleport start(.*)_ExecStart=/usr/local/bin/teleport start --fips\1_g" /etc/systemd/system/teleport*.service
```

Comment:
The sed expression can be simplified a little by removing the `(.*)` and `\1` from the end of the match and replace components - sed will just replace what is matched and leave the rest of the line alone without it.

Instead I would match the start of the line in a group so you don't need to repeat it:

    sed -i -E 's_^(ExecStart=/usr/local/bin/teleport start)_\1 --fips_' /etc/systemd/system/teleport*.service

Also the `g` modifier on the end doesn't make much sense as this string will not match multiple times on a single line.